test/test_fed_dbm.cpp

```diff
@@ -31,451 +31,451 @@ using namespace std;
 using namespace dbm;
 
 // Range for DBM generation
-#define MAXRANGE 10000
-#define RANGE()  ((rand() % MAXRANGE) + 10)
+constexpr auto MAXRANGE = 10000;
+static inline auto RANGE() { return ((rand() % MAXRANGE) + 10); }
 
 // Progress
-#define PROGRESS() debug_spin(stderr)
+static inline void PROGRESS() { debug_spin(stderr); }
 
 // DBM generation
-#define GEN(D) dbm_generate(D, dim, RANGE())
+static inline void GEN(raw_t* dbm, cindex_t dim) { dbm_generate(dbm, dim, RANGE()); }
 
 // New DBM
-#define NEW(N) raw_t* N = new raw_t[dim * dim]
+static inline raw_t* NEW(cindex_t dim) { return new raw_t[dim * dim]; }
 
 // Delete DBM
-#define FREE(N) delete[] N
+static inline void FREE(raw_t* ptr) { delete[] ptr; }
```

Comment:
1) C (dbm provide C API) operate on raw pointers and cannot have smart pointers. The result is a massive mess with ownership.
2) This was just a quick way of getting rid of macros (I did not want to break it with non-trivial changes, because there are way too many of them).

The whole library needs redesign: C library with extra C++ API makes no sense; it needs to be the opposite: C++ library with extra C API if any.

Reply:
the new and free macros where only used at the start and end of the test scope. Which is the same a smart pointer would do, then when we want to use the value we just do .get or dereference.

But you can just leave it be, its probably not worth spending more time on a test